Mathlib/Probability/Kernel/Composition.lean

```diff
@@ -689,39 +695,72 @@ def prodMkLeft (γ : Type*) [MeasurableSpace γ] (κ : kernel α β) : kernel (
   comap κ Prod.snd measurable_snd
 #align probability_theory.kernel.prod_mk_left ProbabilityTheory.kernel.prodMkLeft
 
+/-- Define a `kernel (α × γ) β` from a `kernel α β` by taking the comap of the projection. -/
+def prodMkRight (κ : kernel α β) (γ : Type*) [MeasurableSpace γ] : kernel (α × γ) β :=
+  comap κ Prod.fst measurable_fst
```

Comment:
The idea was that if you add a type to the left, the type argument is to the left of the kernel. And if you add a type to the right, the type argument is to the right.

Reply:
Example in PFR: `(prodMkRight κ T') ×ₖ (prodMkLeft T η)` where `{κ : kernel T G} {η : kernel T' G}`. I thought placing the type to one side or the other helped visually understand what we get.

Reply:
Yeah, I understand that, but it goes against my expectation that type arguments go before. Even without this consideration, it seems pretty annoying that `prodMkLeft` and `prodMkRight` have different argument orders even though they are "the same operation", eg when writing an API for both you will need to swap the arguments to half the calls because of this.

Reply:
Since it really does not matter much: switched.